### PR TITLE
Setup Github Actions with Docker for Testing and Coverage

### PR DESCRIPTION
**Pull request by nateglims**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    libssl-dev \
+    lcov \
+    ruby \
+    wget
+
+# get an appropriate version of cmake
+RUN wget -qO- "https://cmake.org/files/v3.18/cmake-3.18.1-Linux-x86_64.tar.gz" | tar --strip-components=1 -xz -C /usr/local
+
+# set up certs for testing
+RUN mkdir /certs
```
Comment (aggarw13) [inline Dockerfile:16]: These can be scoped down to the build only for the `integtest`?

Comment (nateglims) [reply Dockerfile:16]: That's a good idea. I was going to take a stab at getting demos working through actions now (many using certs) and I'm still working out the best way to keep these understandable while also reducing repetition.

**Comment by nateglims** [comment]
test comment

File: CMakeLists.txt [hunk 2]
```diff
@@ -28,7 +28,10 @@ include( "demos/logging-stack/logging.cmake" )
 
 # Configure options to always show in CMake GUI.
 option( BUILD_TESTS
-        "Set this to ON to build both demo and test executables. When OFF, only demo executables are built."
+        "Set this to ON to build both demo and unit test executables. When OFF, only demo executables are built."
```
Comment (sarenameas) [inline CMakeLists.txt:31]: When OFF, only demo and/or integration test executables are built." ?

**Pull request closed without merging**